src/Integrations/OpenAi/ProductMapper.php

```diff
@@ -979,101 +964,117 @@ private function get_shipping(): string {
-		$country = '';
-		$region  = '';
-
-		switch ( $location->type ) {
-			case 'country':
-				$country = $location->code;
-				break;
-			case 'state':
-				list($country, $region) = array_pad( explode( ':', $location->code ), 2, '' );
-				break;
-			case 'continent':
-				$country = $location->code;
-				break;
-			default:
-				return null;
+		// There is the "Locations not covered by other zones" zone.
+		$generic_zone = WC_Shipping_Zones::get_zone( 0 );
+		if ( ! $generic_zone || is_wp_error( $generic_zone ) ) {
+			$this->cached_shipping_zones = [];
+			return $this->cached_shipping_zones;
 		}
 
-		$parts = array_filter( [ $country, $region, $method_title ] );
-
-		if ( '' !== $price ) {
-			$parts[] = sprintf( '%s %s', $price, $currency );
-		}
-
-		return implode(
-			':',
-			array_map(
-				function ( $part ) {
-					return trim( $part, ':' );
-				},
-				$parts
-			)
-		);
+		$this->cached_shipping_zones = [
+			[
+				'zone_locations'   => [
+					(object) [
+						'type' => 'not_covered',
+						'code' => '',
+					],
+				],
+				'shipping_methods' => $generic_zone->get_shipping_methods(),
+			],
+		];
+		return $this->cached_shipping_zones;
 	}
 
 	/**
-	 * Check if local pickup is available (cached to prevent repeated zone queries)
+	 * Get shipping data from WooCommerce zones (cached globally to prevent repeated queries).
 	 *
-	 * @return bool True if local pickup is available.
+	 * Format: country:region:service_class:price
+	 * Example: US:CA:Overnight:16.00 USD:BG:VAR:Flat rate:10.00 USD
+	 * Note: All 4 parts are required, even if region is empty (e.g., US::Flat rate:10.00 USD).
+	 * Note: Colons in service_class are escaped with backslash (e.g., Express\: Next Day).
+	 * Important: We are assuming that an empty string can be used for both country and region.
+	 *            This assumption might be completely wrong.
+	 *
+	 * @return string Shipping data string.
 	 */
-	private function has_local_pickup(): bool {
-		if ( null !== self::$cached_has_local_pickup ) {
-			return self::$cached_has_local_pickup;
-		}
-
-		if ( ! class_exists( 'WC_Shipping_Zones' ) ) {
-			self::$cached_has_local_pickup = false;
-			return self::$cached_has_local_pickup;
+	private function get_shipping(): string {
+		if ( null !== $this->cached_shipping_data ) {
+			return $this->cached_shipping_data;
 		}
 
-		$zones = $this->get_cached_shipping_zones();
+		$shipping_data = [];
+		$zones         = $this->get_cached_shipping_zones();
+		$currency      = $this->get_currency_code();
 
 		foreach ( $zones as $zone ) {
+			$locations = $zone['zone_locations'];
+
 			foreach ( $zone['shipping_methods'] as $method ) {
-				if ( 'local_pickup' === $method->id ) {
-					self::$cached_has_local_pickup = true;
-					return self::$cached_has_local_pickup;
+				// Escape colons in method title.
+				$method_title = trim( str_replace( ':', '\:', $method->get_method_title() ), ':' );
```

Comment:
Nitpick: `trim` method_title before `replace` for the colon.
```suggestion
				$method_title = str_replace( ':', '\:', trim( $method->get_method_title(), ':' ) );
```

Reply:
Done in https://github.com/woocommerce/woocommerce-product-feed-for-openai/pull/35/commits/8defb2a42d3ea4f0f730321c4b865c2a95f76bd3. I also had to switch to `get_title()` so that the correct thing would get returned, so I added tests to make sure it's all handled well.